### PR TITLE
docs: code review guide & pull request FAQ

### PR DESCRIPTION
**Pull request by pfuscoe**
I intended to create a code review guide from both the perspective of the reviewer and the person submitting the code. After working on it a bit, it seemed to make sense to break this into two separate documents from each point of view.

Any feedback is welcome as I think this should be treated as a starting point for discussion. I'd be glad to get some consensus and improve on it further.

### CODE REVIEW

File: docs/code-review-guide.md
Comment (rmutter) [inline docs/code-review-guide.md:15]: Something that is time consuming and we haven't done much lately, but could be something worth considering is pulling down the branch and running the person's code locally. This would not be tenable for someone on the team doing many PRs across many projects but could be a good practice on projects with 2 devs.

File: docs/code-review-guide.md
Comment (rmutter) [inline docs/code-review-guide.md:23]: I think there's room to potentially have a bunch of examples or things we specifically look for each in of these categories. Might be worth trying to crowdsource examples from the rock team / dev team into a document.

**Review by rmutter — APPROVED** [review]

File: docs/code-review-guide.md
Comment (will-t-harris) [reply docs/code-review-guide.md:7]: I'd advocate for leaving it as "Having out work reviewed by peers ensures..."

File: docs/code-review-guide.md
Comment (will-t-harris) [inline docs/code-review-guide.md:13]: There's a third option here, which is to leave a comment instead of approving or requesting changes.

I will sometimes do this when I have questions or comments about a piece of code, and I'd like to hear from the author before deciding whether to request changes or approve.

File: docs/code-review-guide.md
Comment (will-t-harris) [inline docs/code-review-guide.md:29]:
```suggestion
Is the code easy to read and understand? Could a developer who is unfamiliar with the codebase quickly understand what your code does?   

It's important to always think about the potential for new developers that may join later and how quickly they'll be able to contribute.  

One way to think about this: the collection of names in our code should create a vocabulary for talking about the code. If the vocabulary is confusing or inconsistent, then maintaining and changing the code will be more difficult.  
```

File: docs/code-review-guide.md
Comment (will-t-harris) [reply docs/code-review-guide.md:23]: Feels to me like this could be a separate PR or a separate doc.

**Review by will-t-harris — APPROVED** [review]
Looks great Patrick!

Just left a couple suggestions below, but I think this is a good start for sure. 🤝

File: docs/pull-request-faq.md
Comment (dgillen27) [inline docs/pull-request-faq.md:9]: Small nitpick here but I think we might want to say `relevant ticket` instead of `work ticket` 😄

File: docs/pull-request-faq.md
Comment (dgillen27) [inline docs/pull-request-faq.md:33]: ` If the lead developer approves your changes, it's assumed that you can go ahead with the merge.` Is this how we usually go about things? My experience has been that we generally wait for 2 reviews unless the feature is very small/needs to be rushed

**Review by dgillen27 — APPROVED** [review]
Looks great Patrick 🥳 just a few bits of feedback from me

**Review by emcclosky — APPROVED** [review]

File: docs/code-review-guide.md
Comment (pfuscoe) [reply docs/code-review-guide.md:7]: Sounds good - I think leaving it as just 'peers' may be fine

File: docs/code-review-guide.md
Comment (pfuscoe) [reply docs/code-review-guide.md:15]: Sounds good. I'll add a note here about it. It might be worth talking through this a bit, but for now I can phrase along the lines of 'encouraged when possible'

File: docs/code-review-guide.md
Comment (pfuscoe) [reply docs/code-review-guide.md:23]: Makes sense. Leaving myself a to-do on this - maybe we can see if the team has ideas to contribute during the next round-up

File: docs/code-review-guide.md
Comment (pfuscoe) [reply docs/code-review-guide.md:13]: Great point - I'll add a sentence or two on this

File: docs/pull-request-faq.md
Comment (pfuscoe) [reply docs/pull-request-faq.md:33]: I think what I meant to say didn't come across very well here. I was thinking that as long as the lead dev has approved after 24 hours, it's okay to merge.

It feels like we're handling this a bit differently depending on the dev team or project. We should probably discuss this some more

**Pull request merged**
